tests/test_rotations.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import pytest
+from numpy.typing import NDArray
+from scipy.spatial.transform import Rotation
+
+from timemachine.lib import custom_ops
+
+
+def hamiliton_product(q1: NDArray, q2: NDArray) -> NDArray:
+    """
+    Compute the product of two quaternions, called a hamilton product.
+
+    References
+    ----------
+    https://en.wikipedia.org/wiki/Quaternion#Hamilton_product
+    """
+    w1, x1, y1, z1 = q1[0], q1[1], q1[2], q1[3]
+    w2, x2, y2, z2 = q2[0], q2[1], q2[2], q2[3]
+    output = np.empty(4)
+    output[0] = (w1 * w2) - (x1 * x2) - (y1 * y2) - (z1 * z2)
+    output[1] = (w1 * x2) + (x1 * w2) + (y1 * z2) - (z1 * y2)
+    output[2] = (w1 * y2) - (x1 * z2) + (y1 * w2) + (z1 * x2)
+    output[3] = (w1 * z2) + (x1 * y2) - (y1 * x2) + (z1 * w2)
+    return output
+
+
+def rotate_vector_by_quaternions(coords: NDArray, quaternion: NDArray) -> NDArray:
```

Comment:
nit: should the name of this function by `rotate_vector_by_quaternion` (singular)?

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/1174/commits/88e9f274e2b2a18a87869d913a4332d9f011fa53